public/javascript/pump.js

```diff
@@ -669,22 +658,53 @@ if (!window.Pump) {
     };
 
     Pump.ajaxError = function(jqXHR, textStatus, errorThrown) {
-        Pump.error(Pump.jqxhrError(jqXHR));
+        Pump.jqxhrError(jqXHR);
     };
 
     Pump.jqxhrError = function(jqxhr) {
         var type = jqxhr.getResponseHeader("Content-Type"),
             response;
-        if (type && type.indexOf("application/json") !== -1) {
+
+        if (_.isObject(jqxhr.responseJSON) &&
+            _.isString(jqxhr.responseJSON.error)) {
+            Pump.error(new Error(jqxhr.responseJSON.error));
+        } else if (type && type.indexOf("application/json") !== -1) {
             try {
                 response = JSON.parse(jqxhr.responseText);
                 Pump.error(new Error(response.error));
             } catch (err) {
                 Pump.error(new Error(jqxhr.status + ": " + jqxhr.statusText));
             }
-        } else {
+        } else if (jqxhr.status && jqxhr.statusText) {
             Pump.error(new Error(jqxhr.status + ": " + jqxhr.statusText));
```

Comment:
Can't we now get rid of this branch since `Pump.error` will do all the heavy lifting of figuring out what to do with the error?

Reply:
Yes, some logic should be moved, but make sense

Reply:
done: https://github.com/pump-io/pump.io/pull/1352/commits/3d9e3a3b476be975bd47052fbb560f3287c9909e